### ISSUE TITLE
Need some more metrics for Gitlab CI Pipelines

### ISSUE THREAD

**Issue by satish-chef** [mvisonneau/gitlab-ci-pipelines-exporter#232]
Hi @mvisonneau ,

This looks like a great exporter for various Gitlab metrics.  However I was wondering If you could add below metrics also to this prometheus exporter:

**CI Pipeline metrics**:
Whose pipeline failed the most (per person)?
Whose pipeline succeeded the most (per person)?
Number of Test case failures

**Merge request metrics**:

How many merge requests per day, per week, per month and year?
Average review time
Number of requests approved (by person)
Number of requests rejected (by person)
Identify the person who reviewed the code (esp. when one of the stage in pipeline fails)

**Repository based** (per repositiory):
How many commits per day, month, year on development branch?

**Comment by mvisonneau** [comment]
👋 hey @satish-chef,

Assuming all of these things are available through the GitLab API, I believe they could be implemented. PRs are welcome though! :)

**Comment by satish-chef** [comment]
Hi @mvisonneau  - A humble request/reminder to please add these metrics to your tool.

**Comment by satish-chef** [comment]
> 👋 hey @satish-chef,
> 
> Assuming all of these things are available through the GitLab API, I believe they could be implemented. PRs are welcome though! :)

**Comment by satish-chef** [comment]
> Hi @mvisonneau - A humble request/reminder to please add these metrics to your tool.

**Comment by satish-chef** [comment]
Hi @mvisonneau   Just checking if have managed to add the above mentioned metrics in your tool.

**Comment by satish-chef** [comment]
Hi @mvisonneau , hope you are doing fine.  Can you please check and update if the above mentioned metrics have been added to your exporter tool.

**Comment by mvisonneau** [comment]
AFAIK they have not.

**Comment by satish-chef** [comment]
@mvisonneau Yes, you are right, Gitlab hasn;'t exposed these metrics through the Gitlab API.  As per their reponse to my Gitlab issue, it will be done may be in in late 2022.  Gitlab ticket for your reference  - https://gitlab.com/gitlab-org/gitlab/-/issues/338856
